src/nprintml/automl.py

```diff
@@ -0,0 +1,143 @@
+        leaderboard = predictor.leaderboard(test_data, silent=True)
+        leaderboard = leaderboard.set_index('model').sort_index()
+        leaderboard.to_csv(self.outpath / 'leaderboard.scv')
+
+    def generate_graphs(self, predictor, test_data):
+        y_true = test_data['label']
+        test_no_label = test_data.drop(labels=['label'], axis=1)
+        binarizer = LabelBinarizer().fit(y_true)
+        binarized_labels = binarizer.transform(y_true)
+
+        y_pred = predictor.predict(test_no_label)
+        y_proba = predictor.predict_proba(test_no_label)
+        self.make_pr(binarizer.classes_, binarized_labels, y_proba)
+        self.make_roc(binarizer.classes_, binarized_labels, y_proba)
+        self.make_cfmx(binarizer.classes_, y_true, y_pred, normalize='all')
+
+    def make_cfmx(self, classes, y_true, y_pred, normalize=None):
+        cm = confusion_matrix(y_true, y_pred)
+        c = ConfusionMatrixDisplay(cm, display_labels=classes)
+        c.plot(include_values=False, cmap=plt.cm.Blues, xticks_rotation=45)
+
+        self.finalize_graph(self.outpath / 'cfmx.pdf',
+                            x_label='Predicted Label', y_label='True Label')
+
+    def make_pr(self, classes, y_true_bin, y_proba):
+        # Binary case does not enumerate
+        if len(classes) == 2:
+            ax = self._make_binary_pr(y_true_bin.ravel(), y_proba.ravel())
+        else:
+            for i, cl in enumerate(classes):
+                ax = self._make_binary_pr(y_true_bin[:, i], y_proba[:, i], cl=cl)
+
+        # Split up the line styles to make them unique
+        linestyles = ['-', '--', '-.', ':']
+        for i, line in enumerate(ax.lines):
+            line.set_linestyle(linestyles[i % len(linestyles)])
+
+        self.finalize_graph(self.outpath / 'pr.pdf', x_lim=[0.0, 1.0],
+                            y_lim=[0.0, 1.05], legend_loc='lower left',
+                            x_label='Recall', y_label='Precision')
+
+    def _make_binary_pr(self, y_true, y_proba, cl=None):
+        precision, recall, thresholds = precision_recall_curve(y_true, y_proba)
+        cl_ap = average_precision_score(y_true, y_proba)
+        label = None
+        if cl:
+            label = '{0} - AP: {1:.2f}'.format(cl, cl_ap)
+        else:
+            label = 'AP: {0:.2f}'.format(cl_ap)
+        ax = sns.lineplot(x=recall, y=precision, label=label)
+
+        return ax
+
+    def make_roc(self, classes, y_true_bin, y_proba):
+        # Binary case does not enumerate
+        if len(classes) == 2:
+            ax = self._make_binary_roc(y_true_bin.ravel(), y_proba.ravel())
+        else:
+            for i, cl in enumerate(classes):
+                ax = self._make_binary_roc(y_true_bin[:, i], y_proba[:, i], cl=cl)
+
+        # Split up the line styles to make them unique
+        linestyles = ['-', '--', '-.', ':']
+        for i, line in enumerate(ax.lines):
+            line.set_linestyle(linestyles[i % len(linestyles)])
```

Comment:
The `mod` way works great so you might prefer it -- just as an alternative --

```suggestion
        linestyles = ('-', '--', '-.', ':')
        for (linestyle, line) in zip(itertools.cycle(linestyles),
                                     ax.lines):
            line.set_linestyle(linestyle)
```